plugins/woocommerce/src/Database/Migrations/MetaToMetaTableMigrator.php

```diff
@@ -61,33 +61,77 @@ public function __construct() {
 	}
 
 	/**
-	 * Migrate a batch of entities from the posts table to the corresponding table.
+	 * Return data to be migrated for a batch of entities.
 	 *
-	 * @param array $entity_ids Ids of entities ro migrate.
+	 * @param array $entity_ids Ids of entities to migrate.
+	 *
+	 * @return array[] Data to be migrated. Would be of the form: array( 'data' => array( ... ), 'errors' => array( ... ) ).
 	 */
-	protected function process_migration_batch_for_ids_core( array $entity_ids ): void {
+	public function fetch_sanitized_migration_data( $entity_ids ) {
+		$this->clear_errors();
 		$to_migrate = $this->fetch_data_for_migration_for_ids( $entity_ids );
 		if ( empty( $to_migrate ) ) {
-			return;
+			return array(
+				array(),
+				array(),
+			);
```

Comment:
Also optional, because I think wherever the returned array is consumed we have appropriate checks, but, should this ideally be:

```suggestion
			return array(
				'data'   => array(),
				'errors' => array(),
			);
```

Reply:
Thanks for catching, seems like I missed this.

Reply:
Addressed in https://github.com/woocommerce/woocommerce/pull/38993/commits/94fce9c169809e484a9d75803600ae3ad88b4ee5